interfaces/builtin/download.go

```diff
@@ -0,0 +1,325 @@
+     bus=session
+     path=/com/canonical/applications/download/@{SNAP_LABEL_DBUS}/**
+     interface=com.canonical.applications.Download
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive)
+     bus=session
+     path=/com/canonical/applications/download/@{SNAP_LABEL_DBUS}/**
+     interface=org.freedesktop.DBus.Properties
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/com/canonical/applications/download/@{SNAP_LABEL_DBUS}/**
+     interface=com.canonical.applications.GroupDownload
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Be explicit about the allowed members we can send to
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=createDownload
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=createDownloadGroup
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=getAllDownloads
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=getAllDownloadsWithMetadata
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=defaultThrottle
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=isGSMDownloadAllowed
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Explicitly deny DownloadManager APIs apps shouldn't have access to in order
+# to make sure they aren't accidentally added in the future (see LP: #1277578
+# for details)
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=allowGSMDownload,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=createMmsDownload,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=exit,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=setDefaultThrottle,
+`)
+
+var downloadPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as a download manager.
+# Usage: common
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+#include <abstractions/dbus-session-strict>
+
+# https://specifications.freedesktop.org/download-spec/latest/
+# allow binding to the DBus download interface
+dbus (bind)
+    bus=session
+    name="com.canonical.applications.Downloader",
+
+# Allow writing to app download directories
+owner @{HOME}/snap/*/common/Downloads/    rw,
+owner @{HOME}/snap/*/common/Downloads/**  rwk,
```

Comment:
Seems like the download manager should only get access to the directories of snaps that are connected. So this should be a per-slot connection with the name of the package in it.